### PR TITLE
Add a gradio demo and diffusers code reference

### PR DESCRIPTION
**Pull request by apolinario**
Add a Gradio demo that runs with the original pipeline code (incl. image-to-image)

### CODE REVIEW

File: demo_gr.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+                exif_data[ExifTags.Base.Software] = "AI generated;txt2img;flux"
+            else:
+                exif_data[ExifTags.Base.Software] = "AI generated;img2img;flux"
+            exif_data[ExifTags.Base.Make] = "Black Forest Labs"
+            exif_data[ExifTags.Base.Model] = self.model.__class__.__name__
+            if add_sampling_metadata:
+                exif_data[ExifTags.Base.ImageDescription] = prompt
+            img.save(buffer, format="jpeg", exif=exif_data, quality=95, subsampling=0)
+
+            img_bytes = buffer.getvalue()
```
Comment (jenuk) [inline demo_gr.py:153]: You are not using the bytes. Why calculate them? Maybe you could add an option to save generated images? Then you wouldn't get `.webp` images if you hit download in gradio.

**Review by jenuk — CHANGES_REQUESTED** [review]
Can you also add gradio to the requirements in the pyproject-file? Please do it as a separate requirements just as the streamlit requirement? Feel free to add it to `all` too.

If you take the seed of a previous calculation and pass it through the demo again you will get a different output. If you generate another time afterwards it will give you the same image again. I'm assuming that gradio has an internal representation of that seed as a float somewhere in the middle and loses precision. This should be fixed.

**Comment by apolinario** [comment]
> Can you also add gradio to the requirements in the pyproject-file? Please do it as a separate requirements just as the streamlit requirement? Feel free to add it to all too.

Done!

> If you take the seed of a previous calculation and pass it through the demo again you will get a different output. If you generate another time afterwards it will give you the same image again. I'm assuming that gradio has an internal representation of that seed as a float somewhere in the middle and loses precision. This should be fixed.

I found the issue. The issue is that the seed is normally calculated in Python, however when passed to the Javascript front-end, Gradio uses Javascript's `Number`, which has a maximum value of `2^53 - 1` or `9007199254740992` so if the seed is larger than that, it will replace the extra characters with zeroes 😅. The workaround I did is to do `str` in Javascript but convert to `int` for Python that can handle the larger numbers.

**Comment by apolinario** [comment]
I also externalized from the advanced options accordion the image2image option!

**Comment by jenuk** [comment]
Looks good! One last thing: Can you move the jpg that are generated into either a tempdir or into the `output` (or `output/gradio/`) dir that is also used in the other demos? Otherwise the script will spam the root in which it is executed.

**Comment by apolinario** [comment]
Makes sense, done!

**Review by zansun-nusnaz — APPROVED** [review]
Thank you. Great addition!

**Review by jenuk — APPROVED** [review]
Thanks @apolinario for bearing with my requests. Looks top now

**Pull request merged**